dhcpv4/option_class_identifier.go

```diff
@@ -0,0 +1,52 @@
+package dhcpv4
+
+import (
+	"fmt"
+)
+
+// This option implements the Class Identifier option
+// https://tools.ietf.org/html/rfc2132
+
+// OptClassIdentifier represents the DHCP message type option.
+type OptClassIdentifier struct {
+	Identifier string
```

Comment:
should this be `string` or `[]byte` ? Looking at the RFC I'd say `[]byte`, then all the string conversions should be done elsewhere

Reply:
I was going to make the specific opts more API/human-friendly, then handle serialization in the ToBytes method.

Reply:
I think that the API should be as friendly as possible, as long as it doesn't create other kinds of risk. The string could be a multi-byte string when decoded, which may cause issues with `Length`, and the decoding of all the following options as a consequence

Reply:
as you showed me offline with https://play.golang.org/p/sXdfZg0987B, your code is good, and I learned something new, thanks!